addOns/commonlib/src/test/java/org/zaproxy/addon/commonlib/DiceMatcherUnitTest.java

```diff
@@ -132,4 +136,22 @@ void shouldGiveCorrectPercentageForDifferentString() {
         // Then
         assertThat(sim, is(equalTo(25)));
     }
+
+    static Stream<Arguments> stringSetSource() {
+        return Stream.of(
+                Arguments.of(null, ORIGINAL_STRING),
+                Arguments.of(ORIGINAL_STRING, null),
+                Arguments.of(null, null),
+                Arguments.of("ab", ORIGINAL_STRING),
+                Arguments.of(ORIGINAL_STRING, "ab"));
```

Comment:
"ab" -> "a"?
We are only checking if a parameter is a single character ([L59](https://github.com/zaproxy/zap-extensions/pull/2907/files#diff-1ff4022fe74aa6c1d61fc6c935f1c76ef91fb39849f300b9d930698496b4c35eR59)). This test case shouldn't have passed...

Reply:
It probably still calculates to zero, but you're right. For the intent it should be a single char.

Reply:
Fix pushed